pkg/kv/kvserver/replica_proposal_buf.go

```diff
@@ -1444,6 +1459,15 @@ func (rp *replicaProposer) rejectProposalWithRedirectLocked(
 	rangeDesc := r.descRLocked()
 	storeID := r.store.StoreID()
 	r.store.metrics.LeaseRequestErrorCount.Inc(1)
+	if redirectTo == roachpb.ReplicaID(raft.None) {
+		// TODO(pav-kv): should we pass the Lease? Below, however, we return a
+		// "speculative" Lease, so a Lease{} can be considered an edge case of it.
```

Comment:
I think this seems fine. A `Lease{}` indicates that this node has no idea who the current leaseholder nor leader is, and we use that signal elsewhere in the system (e.g. in DistSender circuit breakers) to determine whether the replica is connected to the quorum.

Reply:
SGTM. Changed the TODO into a note.